rust/arrow/src/compute/kernels/arithmetic.rs

```diff
@@ -197,8 +263,13 @@ where
         + Sub<Output = T::Native>
         + Mul<Output = T::Native>
         + Div<Output = T::Native>
-        + Zero,
+        + Zero
+        + One,
 {
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    return simd_divide(&left, &right);
```

Comment:
Is it more readable to use the 
`if cfg!`
pattern?

Reply:
Follow up JIRA is [here](https://issues.apache.org/jira/browse/ARROW-6408?filter=-2).